### PR TITLE
refactor: Convert UsageAnalytics to Kotlin

### PR DESCRIPTION
**Pull request by david-allison**
Note: tests have changed a lot, needs a thorough review

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
WARN: Conflicts with #10850

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt [hunk 1]
```diff
@@ -236,207 +231,227 @@ public class UsageAnalytics {
...
 
-
-    public static Throwable getCause(Throwable t) {
-        Throwable cause;
-        Throwable result = t;
-
-        while (null != (cause = result.getCause()) && (!result.equals(cause))) {
-            result = cause;
+    @JvmStatic
+    @KotlinCleanup("convert to ")
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt:262]: That is quite the open-ended statement ;-)

**Review by mikehardy — APPROVED** [review]
LGTM with one comment on a KotlinCleanup comment.
Happy to see auto-merge --> go after addressing that

**Comment by david-allison** [comment]
> LGTM with one comment on a KotlinCleanup comment. Happy to see auto-merge --> go after addressing that

`codecov` is blocking automerge

**Comment by mikehardy** [comment]
> > LGTM with one comment on a KotlinCleanup comment. Happy to see auto-merge --> go after addressing that
> 
> `codecov` is blocking automerge

that only happens when you need to re-run as it's already failed and doesn't update until next results come in. :thinking:

**Comment by mikehardy** [comment]
passed testing locally

**Pull request merged**